iam/snippets/src/test/java/RoleIT.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.cloud.iam.admin.v1.IAMClient;
+import com.google.iam.admin.v1.DeleteRoleRequest;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.UUID;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+
+public class RoleIT {
+  private ByteArrayOutputStream bout;
+
+  private static final String projectId = System.getenv("IAM_PROJECT_ID");
+  private static final String GOOGLE_APPLICATION_CREDENTIALS = System.getenv("IAM_CREDENTIALS");
+  private static final String _suffix = UUID.randomUUID().toString().substring(0, 6);
+  private static final String roleId = "testRole" + _suffix;
+  private static final String roleName = "projects/" + projectId + "/roles/" + roleId;
+
+  private static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @Rule public Timeout globalTimeout = Timeout.seconds(300); // 5 minute timeout
+
+  @BeforeClass
+  public static void checkRequirements() throws IOException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+
+    requireEnvVar("IAM_PROJECT_ID");
+    requireEnvVar("IAM_CREDENTIALS");
+
+    stdOut.close();
+    System.setOut(out);
+  }
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(bout);
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+  }
+
+  @AfterClass
+  public static void cleanUp() {
+    try (IAMClient iamClient = IAMClient.create()) {
+      iamClient.deleteRole(DeleteRoleRequest.newBuilder().setName(roleName).build());
+    } catch (NotFoundException e) {
+      System.out.println("Role no longer exists already.");
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Test
+  public void testRole() {
```

Comment:
Can we break all of this into their own individual test cases instead of all being in one?

Reply:
why do we do that? `@Before`, `@After`, `@BeforeClass`, `@AfterClass` let us achieve test separation. i've also found lots of precedences.

- https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/iam/snippets/src/test/java/DenyIT.java
- https://github.com/googleapis/java-pubsub/tree/main/samples/snippets/src/test/java/pubsub

Reply:
Im leaning towards having it broken in light of making it easy for future contributors not having to worry about samples they didn't touch. 

@averikitsch and @lesv to chime in what would be the best direction to take here.

Reply:
This repo does have technical debt so precedence should not outweigh a best practice. It is best practice to keep tests more isolated; however, with these samples being highly scoped, I don't think there's a high risk of keeping the test this way. Following best practices here may double the size of this file and make maintenance harder.